ephios/plugins/questionnaires/models.py

```diff
@@ -0,0 +1,162 @@
+            "serializer_kwargs": {},
+        }
+
+        field["form_kwargs"] = {
+            "label": self.question_text,
+            "help_text": self.description,
+            "initial": initial,
+            "required": required,
+        }
+
+        field["serializer_kwargs"] = {
+            "required": required,
+        }
+
+        match self.type:
+            case self.Type.TEXT:
+                field["form_class"] = forms.CharField
+                field["serializer_class"] = serializers.CharField
+                # pylint: disable=protected-access
+                max_length = Answer._meta.get_field("answer").max_length
+                field["form_kwargs"]["max_length"] = max_length
+                field["serializer_kwargs"]["max_length"] = max_length
+            case self.Type.SINGLE:
+                field["form_class"] = forms.ChoiceField
+                field["serializer_class"] = serializers.ChoiceField
+                field["form_kwargs"]["widget"] = (
+                    forms.RadioSelect if len(self.choices) <= 5 else Select2Widget
+                )
+            case self.Type.MULTIPLE:
+                field["form_class"] = forms.MultipleChoiceField
+                field["serializer_class"] = serializers.MultipleChoiceField
+                field["form_kwargs"]["widget"] = Select2MultipleWidget
+
+        if self.type != self.Type.TEXT:
+            assert isinstance(
+                self.choices, list
+            ), f"The choices of question {self.name} are not a list"
+            # pylint: disable=not-an-iterable
+            choices = [(choice, choice) for choice in self.choices]
+            field["form_kwargs"]["choices"] = choices
+            field["serializer_kwargs"]["choices"] = choices
+
+        return field_name, field
+
+    def get_form_slug(self):
+        return "questionnaires_" + slugify(f"{self.pk} {self.name}")
+
+    @staticmethod
+    def get_pk_from_slug(slug: str):
+        """
+        Returns the questions' pk from a form field slug  generated by `Question.get_form_slug`.
+        These slugs have the format `questionnaires_123-demo-question`.
+
+        For this slug, this method would return `123`.
+        """
+        return int(slug.split("_", maxsplit=1)[1].split("-")[0])
+
+
+class Questionnaire(models.Model):
+    shift = models.OneToOneField(Shift, on_delete=models.CASCADE)
+    questions = models.ManyToManyField(Question, blank=True)
+
+    class Meta:
+        verbose_name = _("Questionnaire")
+        verbose_name_plural = _("Questionnaires")
+
+
+class Answer(models.Model):
+    participation = models.ForeignKey(AbstractParticipation, on_delete=models.CASCADE)
+    question = models.ForeignKey(Question, on_delete=models.PROTECT)
+    answer = models.CharField(max_length=100, verbose_name=_("Answer"))
+
+    class Meta:
+        verbose_name = _("Answer")
+        verbose_name_plural = _("Answers")
+
+    def __str__(self):
+        return str(self.answer)
```

Comment:
please use more context (question and participation). This is useful when registering the models for the admin view (which you should please do by adding the following to admin.py in the questionnaires folder):

```
from django.contrib import admin

from ephios.plugins.questionnaires.models import Questionnaire, Question, Answer, SavedAnswer

admin.site.register(Questionnaire)
admin.site.register(Question)
admin.site.register(Answer)
admin.site.register(SavedAnswer)
``